apps/simple-caching/index.js

```diff
@@ -0,0 +1,31 @@
+import { parse, resolve } from 'url'
+
+/*
+  This is the site we're going to proxy. You can put it in your `.fly.yml` file and/or do `fly secret set HOST https://my-uncached-website.com` in the command line after installing `@fly/fly`
+*/
+const destination = app.config.HOST
+
+fly.http.respondWith(async (req) => {
+  /* This just takes apart and reconstructs the url. Not super useful in this example, but I mean to use the params object later to do some other fly.io magic. Also protects against stuff like HOST having or not having a backslash. */
+  const params = parse(req.url)
+  const path = resolve(destination, params.path)
```

Comment:
These two lines can be done mostly like this, no external dependencies required:

```javascript
const url = new URL(req.url)
const path = url.pathname
const backend = new URL(url)

backend.scheme = "https:"
backend.port = 443
backend.hostname = "onehostname.com"
```

There's a relatively advanced proxy function here we'll probably build in sometime: https://github.com/superfly/onehostname/blob/master/src/backends.ts#L140-L187